### ISSUE TITLE
Improve ellipses (...) around highlight snippets in descriptions of cards

### ISSUE THREAD

**Issue by eaquigley** [IQSS/dataverse#537]
---

Author Name: **Philip Durbin** (@pdurbin)
Original Redmine Issue: 3952, https://redmine.hmdc.harvard.edu/issues/3952
Original Date: 2014-05-07
Original Assignee: Philip Durbin

---

Moved from #3887:

The dataset description snippets that are displayed to highlight matching search terms need better render logic to display the "..." at the beginning and end. (Currently they're displaying before and after all snippets.)

**Comment by eaquigley** [comment]
Discussing during 7/7 meeting. Was brought up by @mheppler

**Comment by pdurbin** [comment]
> Discussing during 7/7 meeting

Right. Here are my notes from that meeting that explain some of the problems: https://docs.google.com/document/d/1p8zXIbzlACxfFhumkZN0_niyOM7V5LR8j_x3cOxkboE/edit?usp=sharing

My plan is to spend an hour poking at the code to see what can be done and estimate better the scope of work.

**Comment by pdurbin** [comment]
I started [some notes](https://docs.google.com/document/d/1Xfxppo3QPnfDn61MF-5fbI6-8zr_mx69kwPR5GmLBKg/edit?usp=sharing) as I looked into this issue. I've afraid it's a non-trivial change to dig into all the highlighting code, which could use some cleanup. Also, #1589 is related so we might want to address both issues in the same dev cycle (or not).

The crux of this issue seems to be knowing the following:

- is the snippet complete or a fragment?
- if not a fragment, is the snippet at the beginning, middle, or end?
- are there multiple snippets for more than one match?

I started some of this conversation at http://lucene.472066.n3.nabble.com/boolean-for-if-highlighting-snippet-contains-complete-original-value-or-is-truncated-td4216220.html

I've been saying this might be two or three days of work (just this ticket). Less if it turns out Solr can answer some of these question for us. @scolapasta and I already agreed to move this out of 4.0.2.
